### DISCUSSION TITLE
How to connect to my Repo using the terminal

### DISCUSSION THREAD

**Discussion by vincej** [community/community#38727] [New to GitHub] [answered]
I have created ssh keys on my server and put the public key on my Github Repo and received the message that I have successfully been authenticated, but "but GitHub does not provide shell access".  This is very confusing because I want to be able to do simple stuff like push/pull etc through the terminal. However I can do these things through my lap top. 

Even more curious is that I cloned my Repo to my server using the facilities within PHPStorm which was authenticated, and I can now push and pull through Storm ... but not my Terminal. Whenever I try with say "git pull" It asks for my username and password, which Git always rejects.

How do I get terminal access to my Repo? What have I missed.

**Comment by airtower-luna** [comment, marked as answer]
> Whenever I try with say "git pull" It asks for my username and password, which Git always rejects.

That sounds like your local configuration contains an HTTPS URL for the repository on GitHub rather than SSH. The URL decides which protocol is used, and naturally your SSH key can only be used with SSH. You can change the URL with [`git remote set-url`](https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emset-urlem), the command should look similar to this:

```
git remote set-url origin git@github.com:vincej/...
```

I assumed here that `origin` is the name of the remote, and you can find the correct URL on the repository page (green "Code" button, select "SSH").

**Reply by vincej** [reply]
You are a Rockstar! Thank you very much - it worked
